2022 CMS Chapter (#3120)
HTTPArchive/almanac.httparchive.org: src/content/en/2022/cms.md

---
#See https://github.com/HTTPArchive/almanac.httparchive.org/wiki/Authors'-Guide#metadata-to-add-at-the-top-of-your-chapters
title: CMS
description: CMS chapter of the 2022 Web Almanac covering CMS adoption, user experience of websites running on CMS platforms, and CMS resource weights.
authors: [sirjonathan]
reviewers: [alexdenning, dknauss, alonkochba, honzasladek]
analysts: [csliva]

> editors: [dknauss]

Reviewer: More of an editor, but in this case (CMS, performance) I have a pretty good working knowledge. I have no review comments of substance to make in this chapter.

  chart_url="https://docs.google.com/spreadsheets/d/e/2PACX-1vQpmp9GvO62PKJmEC_yxnHVx4zuqKwYh68tquM5yZCCrOlZETqbLhu3iuSarRq2n4bW7dRbUoSB8NO2/pubchart?oid=898515378&format=interactive",
  sheets_gid="1599740178",
  sql_file="cms_adoption_by_rank.sql"
  )
}}

According to the dataset, CMSs are used by fewer than 7% of the top 1,000 websites for both desktop and mobile even though 47% of all mobile sites in the dataset use a CMS. A possible explanation for this apparent discrepancy, and the one we offered last year, is that smaller businesses with websites tend to use a popular CMS for their ease of use, and those CMSs are easily identified. However, larger businesses with higher ranked websites tend to have custom CMS solutions built by professional web developers that we can't identify.

Another explanation is that higher ranking sites with more resources allocated to their development are more likely to obfuscate the identity of their CMS for security reasons. It is improbable that more than 90% of the top 1,000 would forgo a CMS entirely and more likely that they just don't show up in our dataset.

> A potentially correlated trend is the adoption of "headless" CMSs and the move to separate content — and the CMS that powers it — from the frontend experience offered to end-users.

Reviewer 2: We use spaceless em-dashes (and yes I hate 'em! But that's the style...)

```suggestion
A potentially correlated trend is the adoption of "headless" CMSs and the move to separate content—and the CMS that powers it—from the frontend experience offered to end-users.
```

  chart_url="https://docs.google.com/spreadsheets/d/e/2PACX-1vQpmp9GvO62PKJmEC_yxnHVx4zuqKwYh68tquM5yZCCrOlZETqbLhu3iuSarRq2n4bW7dRbUoSB8NO2/pubchart?oid=2275923&format=interactive",
  sheets_gid="921264211",
  sql_file="lighthouse_category_scores_per_cms.sql",
  width=600,
  height=559
  )
}}

We can see that the median performance scores for most platforms on mobile are low, ranging from 19 to 35, with Duda at 47 the exception. As Philip Walton cited in 2021, <a hreflang="en" href="https://philipwalton.com/articles/my-challenge-to-the-web-performance-community/">this does not directly imply bad results</a> in mobile field data but does imply that all platforms have room for improvements, especially for low-end devices and network connections similar to those Lighthouse attempts to emulate.

WordPress, Joomla, Drupal, and 1C-Bitrix showed no change from last year's results. Wix dropped from 30% to 29% while the rest showed improvements.

> Desktop show—unsurprisingly given the faster CPUs and netwrok—better numbers with all CMSs seeing a 10-20 point improvements.

Reviewer 3: typo netwrok -> network

```suggestion
Desktop show—unsurprisingly given the faster CPUs and network—better numbers with all CMSs seeing a 10-20 point improvements.
```

Our dataset, based on <a hreflang="en" href="https://www.wappalyzer.com/technologies/cms">Wappalyzer’s definition</a> of a CMS, identified over 270 individual CMSs. Know a CMS that’s missing? <a hreflang="en" href="https://github.com/wappalyzer/wappalyzer/blob/7ac625c34432cb35d01abd683f88d3bfadca4cca/CONTRIBUTING.md">Contribute to Wappalyzer</a>.

Some CMSs in the dataset are open source (e.g., WordPress and Joomla) and some of them are proprietary (e.g., Wix and Squarespace). Some CMSs can be used on “free” hosted or self-hosted plans, and there are also options for using these platforms on higher-tiered plans even at the enterprise level.

The CMS space as a whole is a complex, federated universe of CMS ecosystems, separated and at the same time intertwined.

## CMS adoption

Our analysis throughout this work includes desktop and mobile websites. The vast majority of URLs we looked at are in both datasets, but some URLs are only accessed by desktop or mobile devices. This can cause divergences in the data, and we thus look at desktop and mobile results separately.

{{ figure_markup(
  image="cms-adoption.png",

>   caption="47% of mobile sites were flagged as CMS's in 2022.",

Reviewer 2: For figure captions we just give the title, rather than any interpretation:

```suggestion
  caption="CMS adoption.",
```

Can you similarly update the other figure captions?

  description="Column chart showing the adoption of CMSs over the past 3 years. 45% of desktop websites and 47% of mobile websites are built using a CMS in 20212 For 2021, both desktop and mobile are approximately 45%, while in 2020 they are both 42%.",
  chart_url="https://docs.google.com/spreadsheets/d/e/2PACX-1vQpmp9GvO62PKJmEC_yxnHVx4zuqKwYh68tquM5yZCCrOlZETqbLhu3iuSarRq2n4bW7dRbUoSB8NO2/pubchart?oid=386824637&format=interactive",
  sheets_gid="643278583",
  sql_file="cms_adoption.sql"
  )
}}

> As of June 2022, 44.65% of websites within the desktop dataset are powered by a CMS, indicating a drop of 0.35% [from 2021](../2021/cms#cms-adoption). The mobile dataset shows an increase from 45.54% [in 2021](../2021/cms#cms-adoption) to 47.31% here in 2022. In our estimate, the drop is more likely attributed to minor variances in attribution than an indicator of a downward trend in CMS usage.

Reviewer 2: Can we review the decimal places? We're looking for 2 significant figures.

```suggestion
As of June 2022, 45% of websites within the desktop dataset are powered by a CMS, indicating a slight drop of [from 2021](../2021/cms#cms-adoption). The mobile dataset shows an increase from 46% [in 2021](../2021/cms#cms-adoption) to 47% here in 2022. In our estimate, the drop is more likely attributed to minor variances in attribution than an indicator of a downward trend in CMS usage.
```

It is instructive to compare these numbers with another commonly used dataset, such as <a hreflang="en" href="https://w3techs.com/technologies/history_overview/content_management/all/q">W3Techs</a>, which reported that as of June 2021, 64.6% of websites are created using a CMS, up from 59.2% in June 2020, which is an increase of over 9%.

The deviation between our analysis and W3Techs’ analysis can be explained by a difference in research methodologies, and the definition of what is a CMS.

W3Techs definition is the following: “_Content Management Systems are applications for creating and managing the content of a website. We include all such systems in this category, also systems that are often classified as wikis, blog engines, discussion boards, static site generators, website editors or any type of software that provides website content_.”

> As mentioned previously, Wappalyzer has a stricter definition of a CMS, which excludes some major CMSs which appear in W3Techs reports. You can read more about our definition on the [Methodology](../2021/methodology) page.

Reviewer 2: ```suggestion
As mentioned previously, Wappalyzer has a stricter definition of a CMS, which excludes some major CMSs which appear in W3Techs reports. You can read more about our definition on the [Methodology](../methodology) page.
```

  sql_file="cms_adoption_by_rank.sql"
  )
}}

According to the dataset, CMSs are used by less than 7% of the top 1,000 websites, compared to 47% of the complete dataset of all mobile sites in our analysis. A possible explanation, and the one we offered last year, is that smaller businesses and websites tend to use a CMS due to the ease of use, and the higher ranked websites tend to be built with proprietary solutions by professional web developers.

Another explanation is that for higher ranking sites which tend to have more resources allocated to development, the choice of a CMS is more likely to be obfuscated. It seems improbable that more than 90% of the top 1,000 would forgo a CMS entirely and more likely that they just don’t show up in our dataset.

A potentially correlated trend is the adoption of “headless” and the move towards separating content, and the CMS that powers it, from the frontend experience offered to end-users.

While our confidence in the overall dataset remains high, we’re interested in investigating the adoption by rank dataset further in future editions to see if more can be done to detect CMS usage and improve the overall accuracy of our results.

> ## Most Popular CMSs

Reviewer 2: WE use sentence case for headings, not title case:

```suggestion
## Most popular CMSs
```

  sheets_gid="2087504589",
  sql_file="wordpress_page_builders.sql"
  )
}}

Within WordPress, users often make use of a “page builder” that sits as a layer on top of WordPress and provides an interface for content management. This year, with Wappalyzer’s detection methods improving, we looked at page builder adoption. We discovered that of the WordPress sites attributed to a page builder (approximately 34% of all WordPress sites in our dataset), Elementor and WP Bakery are the clear winners, with Divi, SiteOrigin, and Oxygen trailing behind.

As we see it today, page builders exert significant influence on the performance of a site and, historically, have been anecdotal indicators of poor performance. As one example, our dataset indicates that it’s not uncommon for websites to have multiple page builders installed, adding a significant increase to the resources loaded by a site.

Now that we’re tracking page builder data, we’ll have the opportunity in future editions to evaluate year of year changes in page builder adoption and look for connections in those changes to the overall performance of WordPress as a CMS.

> ## CMS User Experience

Reviewer 2: ```suggestion
## CMS user experience
```

{{ figure_markup(
  image="distribution-cms-page-weight.png",
  caption="Wix delivers a competitive median of 3.5mb on the low end of page weight while Squarespace hits 6.5mb as a median size among a larger distribution.",
  description="Distribution showing the percentile distribution of total mobile page weight among each of the top 5 most adopted CMSs.",
  chart_url="https://docs.google.com/spreadsheets/d/e/2PACX-1vQpmp9GvO62PKJmEC_yxnHVx4zuqKwYh68tquM5yZCCrOlZETqbLhu3iuSarRq2n4bW7dRbUoSB8NO2/pubchart?oid=1236078412&format=interactive",
  sheets_gid="859067552",
  sql_file="page_weight_distribution.sql"
  )
}}

The distribution of page weight in each platform’s percentiles is substantial, probably related to the difference in user content across different web pages, the number of images used, plugins, etc. The smallest pages delivered per platform come from WordPress, a marked improvement over last year’s data, which only sends 673KB for their 10th percentile of visits. The largest pages come from Squarespace, with ~11.4 MB delivered for their 90th percentile of visits, a ~2MB increase over last year’s data.

> ### Page Weight Breakdown

Reviewer 2: ```suggestion
### Page weight breakdown
```

In 2021, contributors from the WordPress community acknowledged the current state of performance, in this <a hreflang="en" href="https://make.wordpress.org/core/2021/10/12/proposal-for-a-performance-team/">proposal</a> to create a performance dedicated core team, which can hopefully improve the current performance of the average WordPress sites.

[Setup this year’s YoY analysis]

### Adoption by geography

First, we examined WordPress adoption by geography, across all sites in our dataset, comparing to 2021 results.

{{ figure_markup(
  image="wordpress-adoption-geo-yoy.png",
  caption="Year over year, less Wordpress sites were discovered across all geographies.",
  description="Bar chart depicting the adoption of WordPress in each of the 10 geographies with the most websites.",
  chart_url="https://docs.google.com/spreadsheets/d/e/2PACX-1vQpmp9GvO62PKJmEC_yxnHVx4zuqKwYh68tquM5yZCCrOlZETqbLhu3iuSarRq2n4bW7dRbUoSB8NO2/pubchart?oid=442489769&format=interactive",
  sheets_gid="694875761",
  sql_file="cms_adoption_by_geo.sql",
  width=600,
  height=579
  )
}}

> [Analysis]

Reviewer 2: @sirjonathan are you planning on writing something here?

### Passing CWVs by geography

Next, let’s look at the amount of WordPress origins with passing Core Web Vitals, but this time, breakdown by geography, for mobile devices and compared to 2021 results.

{{ figure_markup(
  image="wordpress-cwv-yoy.png",
  caption="Geography plays a large role with the spread of passing sites ranging from 10% to 52%.",
  description="Bar chart depicting the adoption of WordPress in each of the 10 geographies with the most websites.",
  chart_url="https://docs.google.com/spreadsheets/d/e/2PACX-1vQpmp9GvO62PKJmEC_yxnHVx4zuqKwYh68tquM5yZCCrOlZETqbLhu3iuSarRq2n4bW7dRbUoSB8NO2/pubchart?oid=1279742547&format=interactive",
  sheets_gid="1742929367",
  width=600,
  height=579
  )
}}

> [Analysis]

Reviewer 2: @sirjonathan are you planning on writing something here?

### Plugins

We explored how WordPress sites use external resources and separated them between resources that are included in plugins, themes, and shipped in WordPress core (wp-includes), compared to 2021 results.

{{ figure_markup(
  image="median-wordpress-resources.png",
  caption="Wordpress did not change their resources pattern in form or function.",
  description="Column chart showing the percentile distributions of resources used for WordPress websites.",
  chart_url="https://docs.google.com/spreadsheets/d/e/2PACX-1vQpmp9GvO62PKJmEC_yxnHVx4zuqKwYh68tquM5yZCCrOlZETqbLhu3iuSarRq2n4bW7dRbUoSB8NO2/pubchart?oid=660449132&format=interactive",
  sheets_gid="1922458949"
  )
}}

> [Analysis]

Reviewer 2: @sirjonathan are you planning on writing something here?
